---
identifier: "CVE-2020-28975"
package_slug: "pypi/scikit-learn"
title: "Denial of Service"
description: "The `svm_predict_values` in `svm.cpp` in Libsvm, as used in scikit-learn
  and other products, allows attackers to cause a denial of service (segmentation
  fault) via a crafted model SVM (introduced via pickle, json, or any other model
  permanence standard) with a large value in the `_n_support` array. Note, the scikit-learn
  vendor's position is that the behavior can only occur if the library's API is violated
  by an application that changes a private attribute."
date: "2020-12-03"
pubdate: "2020-11-21"
affected_range: "==0.23.2"
fixed_versions:
- "0.24.0"
affected_versions: "Version 0.23.2"
not_impacted: "All versions before 0.23.2, all versions after 0.23.2"
solution: "Upgrade to version 0.24.0 or above."
urls:
- "https://nvd.nist.gov/vuln/detail/CVE-2020-28975"
cvss_v2: "AV:N/AC:L/Au:N/C:N/I:N/A:P"
cvss_v3: "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H"
uuid: "c66d424c-6a24-46a3-9146-f8ebe4c7891a"
cwe_ids:
- "CWE-1035"
- "CWE-937"
identifiers:
- "CVE-2020-28975"
